src/app/services/technical-record/technical-record.service.ts

```diff
@@ -37,6 +37,16 @@ export class TechnicalRecordService {
     return this.getVehicleTechRecordModels(vin, SEARCH_TYPES.VIN);
   }
 
+  isUnique(valueToCheck: string, searchType: SEARCH_TYPES) {
+    return this.getVehicleTechRecordModels(valueToCheck, searchType).pipe(
+      map(vehicleTechRecord => {
+        const allTechRecords = vehicleTechRecord.flatMap(record => record.techRecord);
+        const isUnique = allTechRecords.length > 0 ? allTechRecords.every(record => record.statusCode === StatusCodes.ARCHIVED) : true;
+        return isUnique;
+      })
```

Comment:
`every()` will not run its predicate on empty slots. So, you can simplify as below. Your unit tests should still pass.
```suggestion
      map(vehicleTechRecord => vehicleTechRecord
        .flatMap(record => record.techRecord)
        .every(record => record.statusCode === StatusCodes.ARCHIVED)
      )
```

Reply:
Think this is outdated now as well, Tom's isUnique function has been brought into this branch now